pkg/router/router_test.go

```diff
@@ -224,6 +225,7 @@ func TestEdgeCaseRoutes(t *testing.T) {
 		WithRoute(route.Declare(http.MethodGet, "/odd///path"), okHandler("odd")),
 		HandleRoute(route.Declare(http.MethodGet, "/reject", route.WithMiddleware(reject)), okHandler("never")),
 	)
+	r.HandleFunc(http.MethodGet, "/not/implemented", nil)
```

Comment:
I know it's a total pain, but for this specific case the behavior should be tested with all four handling signatures (`Handle`/`HandleFunc`, `HandleRoute`/`HandleRouteFunc`). This should resolve the coverage and also ensure we don't get caught by surprise by this again.